### PR TITLE
fix(sql-rule): disabling the rule 'allow-execute-sql-types'  does not work

### LINKED ISSUE #1190: [Bug]: After the sql window check specification is closed, the sql check specification is still triggered.
### ODC version

ODC4.2.3

### OB version

independent

### What happened?

After the sql window check specification is closed, the sql check specification is still triggered.
image

### What did you expect to happen?

After the SQL window inspection specification is closed, the execution results are directly displayed.

### How can we reproduce it (as minimally and precisely as possible)?

chat privately

### Anything else we need to know?

_No response_

### Cloud

_No response_

### PR DESCRIPTION
**Pull request by MarkPotato777**
#### What type of PR is this?

type-bug

#### What this PR does / why we need it:
Disabling the rule 'allow-execute-sql-types'  does not work. The root cause is that it doesn't distinguish from empty properties and disabled properties.

#### Which issue(s) this PR fixes:

Fixes #1190

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: server/odc-service/src/main/java/com/oceanbase/odc/service/regulation/ruleset/SqlConsoleRuleService.java [hunk 1]
```diff
@@ -128,7 +128,7 @@ public <T> Optional<List<T>> getListProperties(@NonNull Long rulesetId, @NonNull
                 return true;
             }).map(rule -> {
                 if (Objects.isNull(rule.getProperties())) {
-                    return Collections.EMPTY_LIST;
+                    return null;
```
Comment (yhilmare) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/regulation/ruleset/SqlConsoleRuleService.java:131]: why null, does empty list not work?

Comment (MarkPotato777) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/regulation/ruleset/SqlConsoleRuleService.java:131]: we need to distinguish 'null' and 'empty list'. 'Null' means that this rule is disabled while 'empty list' means that this rule is enabled but the value is empty.

Comment (yhilmare) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/regulation/ruleset/SqlConsoleRuleService.java:131]: okay

**Review by yhilmare — APPROVED** [review]
lgtm

**Pull request merged**